### DISCUSSION TITLE
Empty x-cloak attribute respected?

### DISCUSSION THREAD

**Discussion by gaslit** [alpinejs/alpine#879] [1. Help]
Hi friends, have a boggle here.

```
a.html
<div  x-data="{ open: false }" class="relative ml-3">
  <div>
    <button x-on:click="open = true" class="flex items-center max-w-xs text-sm text-white rounded-full focus:outline-none focus:shadow-solid">
      <img class="w-8 h-8 rounded-full" src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80" alt="">
    </button>
  </div>

  <div x-cloak="" x-show.transition="open" x-on:click.away="open = false"  class="absolute right-0 w-48 mt-2 origin-top-right rounded-md shadow-lg">
    <div class="py-1 bg-white rounded-md shadow-xs">
      <a href="#" class="block px-4 py-2 text-sm text-gray-700 hover:bg-gray-100" role="menuitem">Your Profile</a>
      <a href="#" class="block px-4 py-2 text-sm text-gray-700 hover:bg-gray-100" role="menuitem">Settings</a>
      <a href="#" class="block px-4 py-2 text-sm text-gray-700 hover:bg-gray-100" role="menuitem">Sign out</a>
    </div>
  </div>
</div>
```

```
a.css (Using TailwindCSS)

@tailwind base;
@tailwind components;
@tailwind utilities;

[x-cloak=""] {
  @apply hidden !important;
}
```

I intend to create a default element with an attrib value of an empty string for x-cloak so I can apply it throughout my project as per this tutorial:

[](https://dev.to/hus_hmd/alpine-js-responsive-x-cloak-3blp)

The idea is to hide the DOM to prevent the momentary flash on a page load event.

But I am unable to make it work. Am i missing out on anything?

Thank you.

**Comment by KevinBatdorf** [comment]
That should work. I just tested it on CodePen. Maybe move the rule to the top of your css file? 

I will typically inline that in the `<head>` by itself. Maybe try that?
```html
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover">
  <style>[x-cloak] { display: none !important; }</style>
```

**Reply by gaslit** [reply]
Hi Kevin, thanks for chiming in. But here x-cloak is defined without an empty attrib and will be applied to all elements containing the x-cloak attribute.

Here I am trying to define a default attribute of x-cloak which has an empty parameter and so can be applied to elements to define a default behavior.

**Reply by ryangjchandler** [reply]
`x-cloak` doesn't accept any values, it acts as a "boolean" attribute, so the way that @KevinBatdorf is suggesting is the correct / recommended way.

**Reply by KevinBatdorf** [reply]
> ...will be applied to all elements containing the x-cloak attribute

Would you want it any other way?

**Reply by gaslit** [reply]
> `x-cloak` doesn't accept any values, it acts as a "boolean" attribute, so the way that @KevinBatdorf is suggesting is the correct / recommended way.

Thanks so much. So Kevin and Ryangjchandler are you saying that the example in 
https://dev.to/hus_hmd/alpine-js-responsive-x-cloak-3blp

is erroneous as x-cloak is strictly boolean and alpinejs evaluates it based on its presence rather than its value.

Here I am trying to define a default attribute of x-cloak which has an empty parameter and so can be applied to elements to define a default behavior.

**Reply by KevinBatdorf** [reply]
Alpine will remove `x-cloak` attributes regardless. That article is demonstrating how to apply different initial styles depending on any extra conditions you want to add. Essentially you may define the component pre-init state before Alpine does anything.

`display:none` is just the obvious use case.

You could also have this if you want. Alpine will still remove it though.
```css
[x-cloak="red"] {
    background: red;
}
```
